finder-app/finder.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+if [ ! $# -eq 2 ]
+then
+	echo "You must provide two arguments. First is the folder and second the string to be searched."
+	exit 1
+fi
+
+if [ ! -d $1 ]
+then
+	echo "$1 is not a directory."
+	exit 1
+fi
+
+#Expression based on the following reference: https://stackoverflow.com/questions/9157138/recursively-counting-files-in-a-linux-directory
+NUMBER_OF_FILES=$(find $1 -type f | wc -l)
+
+#Expression based on the following reference: https://stackoverflow.com/questions/16956810/how-to-find-all-files-containing-specific-text-string-on-linux
+LINE_COUNT=$(grep -rn $1 -e $2 | wc -l)
```

Comment:
@jordicrosmompart consider putting your arguments into variable names for readability and also to decouple argument setup from logic.